cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,207 @@
+package nmstate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	sriovtestclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnetwork "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+	nmstateshared "github.com/nmstate/kubernetes-nmstate/api/shared"
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/api/v1"
+	client "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/networks"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var sriovclient *sriovtestclient.ClientSet
+
+const testNamespace string = "nmstate-sriov-testing"
+
+const nmstateDesiredStateTemplateStaticIP string = `interfaces:
+- name: %s
+  description: test for SR-IOV network operator integration
+  type: ethernet
+  state: up
+  ipv4:
+    dhcp: false
+    address:
+    - ip: 192.0.2.2
+      prefix-length: 24
+    enabled: true`
+
+func init() {
+	sriovclient = sriovtestclient.New("")
+}
+
+var _ = Describe("[nmstate] SR-IOV Network Operator Integration", func() {
+
+	var sriovInfos *cluster.EnabledNodes
+
+	BeforeEach(func() {
+		if !networks.IsSriovOperatorInstalled() {
```

Comment:
does it make sense to check for knmstate too?

Reply:
sounds good